lib/ffmpeg/qsv/transcoder.py

```diff
@@ -151,7 +151,7 @@ def validate_caps(self):
         "Missing one or more required ffmpeg elements: {}".format(list(unmet)))
 
   def gen_input_opts(self):
-    opts = "-init_hw_device qsv=hw:/dev/dri/renderD128 -filter_hw_device hw"
+    opts = "-init_hw_device qsv=hw:" + self.renderDevice + " -filter_hw_device hw"
```

Comment:
Use format specifier instead...
```python
opts = "-init_hw_device qsv=hw:{renderDevice} -filter_hw_device hw"
```

Reply:
I don't understand your statement

Reply:
hmm... are you saying ffmpeg ignores the bad device name (assuming it does not exist) and falls back to something that works?

Reply:
if so, are there any warnings in the console verbose output that we need to check?

Reply:
Looks like no warnings are emitted that we can validate against ☹️ .  Only ffmpeg-vaapi generates command error for bad device name... the other plugins (ffmpeg-qsv, gst-vaapi, gst-msdk) quietly fallback to a working device.  Can you file issues with upstream for each plugin so we can start discussion with developers?  If they want to keep fallback, they at least need to emit a warning so we can check for it in the test suite and fail.  Additionally, we can add check in .slashrc to assert the device node exists and abort the entire run if not.